pkg/config/remote/service/core_agent_stream.go

```diff
@@ -0,0 +1,145 @@
+package service
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"sync"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/DataDog/datadog-agent/pkg/api/security"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/config/remote/service/tuf"
+	"github.com/DataDog/datadog-agent/pkg/proto/pbgo"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type coreAgentStream struct {
+	stream                       pbgo.AgentSecure_GetConfigUpdatesClient
+	streamCtx                    context.Context
+	cancel                       context.CancelFunc
+	streamCancel                 context.CancelFunc
+	agentClient                  pbgo.AgentSecureClient
+	mx                           sync.Mutex
+	currentConfigSnapshotVersion uint64
+	tufClient                    *tuf.DirectorPartialClient
+}
+
+func newCoreAgentStream() (*coreAgentStream, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	token, err := security.FetchAuthToken()
+	if err != nil {
+		cancel()
+		err = fmt.Errorf("unable to fetch authentication token: %w", err)
+		log.Infof("unable to establish stream, will possibly retry: %s", err)
+		return nil, err
+	}
+
+	streamCtx, streamCancel := context.WithCancel(
+		metadata.NewOutgoingContext(ctx, metadata.MD{
+			"authorization": []string{fmt.Sprintf("Bearer %s", token)},
+		}),
+	)
+
+	creds := credentials.NewTLS(&tls.Config{
+		InsecureSkipVerify: true,
```

Comment:
I noticed we didn't introduce this now - it was used before in `NewGRPCSubscriber`.

I don't know how the `authorization` with the tokens works here so I have no opinions on this right now 🙂